modules/cache/redis-enterprise/main.bicep

```diff
@@ -0,0 +1,232 @@
+param roleAssignments array = []
+
+@description('Optional. Tags of the resource.')
+param tags object = {}
+
+@allowed([
+  '1.0'
+  '1.1'
+  '1.2'
+])
+@description('Optional. Requires clients to use a specified TLS version (or higher) to connect.')
+param minimumTlsVersion string = '1.2'
+
+@description('Optional. The size of the Redis Enterprise Cluster. Defaults to 2. Valid values are (2, 4, 6, ...) for Enterprise SKUs and (3, 9, 15, ...) for Flash SKUs.')
+param capacity int = 2
+
+@allowed([
+  'EnterpriseFlash_F1500'
+  'EnterpriseFlash_F300'
+  'EnterpriseFlash_F700'
+  'Enterprise_E10'
+  'Enterprise_E100'
+  'Enterprise_E20'
+  'Enterprise_E50'
+])
+@description('Optional. The type of Redis Enterprise Cluster to deploy.')
+param skuName string = 'Enterprise_E10'
+
+@description('Optional. When true, the cluster will be deployed across availability zones.')
+param zoneRedundant bool = true
+
+@description('Optional. Configuration details for private endpoints. For security reasons, it is recommended to use private endpoints whenever possible.')
+param privateEndpoints array = []
+
+@description('Optional. The databases to create in the Redis Cache Enterprise Cluster.')
+param databases array = []
+
+@description('Optional. The name of the diagnostic setting, if deployed. If left empty, it defaults to "<resourceName>-diagnosticSettings".')
+param diagnosticSettingsName string = ''
+
+@description('Optional. Resource ID of the diagnostic storage account. For security reasons, it is recommended to set diagnostic settings to send data to either storage account, log analytics workspace or event hub.')
+param diagnosticStorageAccountId string = ''
+
+@description('Optional. Resource ID of the diagnostic log analytics workspace. For security reasons, it is recommended to set diagnostic settings to send data to either storage account, log analytics workspace or event hub.')
+param diagnosticWorkspaceId string = ''
+
+@description('Optional. Resource ID of the diagnostic event hub authorization rule for the Event Hubs namespace in which the event hub should be created or streamed to.')
+param diagnosticEventHubAuthorizationRuleId string = ''
+
+@description('Optional. Name of the diagnostic event hub within the namespace to which logs are streamed. Without this, an event hub is created for each log category. For security reasons, it is recommended to set diagnostic settings to send data to either storage account, log analytics workspace or event hub.')
+param diagnosticEventHubName string = ''
+
+@description('Optional. The name of logs that will be streamed. "allLogs" includes all possible logs for the resource. Set to \'\' to disable log collection.')
+@allowed([
+  ''
+  'allLogs'
+  'ConnectionEvents'
+  'audit'
+])
+param diagnosticLogCategoriesToEnable array = [
+  ''
```

Comment:
Not allLogs here?

Reply:
Not supported using the standard method. Was getting failures for it. New audit logging ingestion is being introduced in preview

Reply:
Hm ok. Should it then even be an option? Maybe we can comment it out and add a disclaimer as to why